### DISCUSSION TITLE
num of chunks calculation doesn't consider chunk_size

### DISCUSSION THREAD

**Discussion by kuacakuaca** [rwth-i6/returnn#1342] [General]
is that intended that `chunk_size` is not considered for `nr_of_chunks` and `nr_of_chunks_default`(https://github.com/rwth-i6/returnn/blob/04c14d6a3745f90f0bd10b28ce4d51b1d61afb82/returnn/datasets/basic.py#LL1071C17-L1071C17) comparison?

**Comment by albertz** [comment]
Why should it? Do you think the formula is wrong? I don't think so. What do you think is the correct formula? Can you also show some example input?

E.g., example input: Length 5. Chunk step 3, chunk size 4. You end up with:
```
in:  xxxxx
out: xxxx
        xx00
```
2 chunks in the output. But chunk size is not really relevant. Same with chunk size 3:
```
in:  xxxxx
out: xxx
        xx0
```
(5 - 1) // 3 + 1 = 2 with the current formula.
Or consider length 6, still the same, 2 chunks.
Consider length 7, now you get 3 chunks according to the formula. It should look like:
```
in:  xxxxxxx
out: xxx
        xxx
           x00
```
